wasabi/markdown.py

```diff
@@ -1,36 +1,44 @@
-# coding: utf8
-from __future__ import unicode_literals, print_function
+from typing import Iterable, List, Optional, Sequence
+
+from typing_extensions import Literal
 
 
 class MarkdownRenderer:
     """Simple helper for generating raw Markdown."""
 
-    def __init__(self, no_emoji=False):
+    def __init__(self, no_emoji: bool = False):
         """Initialize the renderer.
 
         no_emoji (bool): Don't show emoji in titles etc.
         """
-        self.data = []
+        self.data: List = []
         self.no_emoji = no_emoji
 
     @property
-    def text(self):
+    def text(self) -> str:
         """RETURNS (str): The Markdown document."""
         return "\n\n".join(self.data)
 
-    def add(self, content):
+    def add(self, content: str):
         """Add a string to the Markdown document.
 
         content (str): Add content to the document.
         """
         self.data.append(content)
 
-    def table(self, data, header, aligns=None):
+    def table(
+        self,
+        data: Iterable[Iterable[str]],
+        header: Sequence[str],
+        aligns: Optional[Sequence[Literal["r", "c", "l"]]] = None,
+    ) -> str:
         """Create a Markdown table.
 
         data (Iterable[Iterable[str]]): The body, one iterable per row,
             containig an interable of column contents.
-        header (Iterable[str]): The column names.
+        header (Sequence[str]): The column names.
```

Comment:
Here, the `Iterable` type causes a `mypy` error:

```
wasabi/markdown.py:45: error: Argument 1 to "len" has incompatible type "Iterable[str]"; expected "Sized"
```

So I changed it into a `Sequence`.

Reply:
I think the `List` is okay for `header`, I was mainly concerned about `data` and whether we potentially had `Iterable` uses for large tables.

Reply:
`Sequence` is fine for `header`, too.